app/controllers/transactions/transaction-download.controller.js

```diff
@@ -22,7 +22,14 @@ const fetchTransactionCsvWithHeader = function fetchTransactionCsvWithHeader (re
     transactionService.logCsvFileStreamComplete(timestampStreamStart, filters, [accountId], req.user, false, req.account.type === 'live')
     res.end()
   }
-  const error = () => renderErrorView(req, res, 'Unable to download list of transactions.')
+  const error = () => {
+    const code = (res || {}).statusCode
+    if (code === 504) {
+      renderErrorView(req, res, 'Your request has timed out. Please apply more filters and try again.')
+    } else {
+      renderErrorView(req, res, 'Unable to download list of transactions.')
+    }
+  }
```

Comment:
I think we decided we weren’t going to change anything for CSV downloads because we decided the chance of encountering a timeout when getting the data for the CSV (after necessarily not encountering a timeout while getting the data for the transactions page previously) was unlikely enough not to bother treating specially.